crates/hyperqueue/src/server/event/log/read.rs

```diff
@@ -1,21 +1,24 @@
+use crate::server::event::log::write::EventChunkHeader;
 use crate::server::event::log::{canonical_header, LogFileHeader};
 use crate::server::event::MonitoringEvent;
 use anyhow::anyhow;
 use flate2::read::GzDecoder;
 use rmp_serde::decode::Error;
 use std::fs::File;
-use std::io::ErrorKind;
+use std::io::{Cursor, ErrorKind, Seek, SeekFrom};
+use std::os::unix::fs::FileExt;
 use std::path::Path;
 
 /// Reads events from a file in a streaming fashion.
 pub struct EventLogReader {
-    source: GzDecoder<File>,
+    source: File,
+    current_chunk: GzDecoder<Cursor<Vec<u8>>>,
```

Comment:
We don't event need to copy the data from the file into a vec, we could just read them directly from the file (I guess). I think that an enum that would store something like `NoChunk`, `ReadingChunk` could be useful here.